guides/Building Your Bot/CreatingACustomMemberGateway.md

```diff
@@ -0,0 +1,171 @@
+Something that is often asked or requested, is a custom member-based gateway; that is per member configs, so you would have different {@link Configuration} values for different guilds.
+
+This is simple to achieve with Klasa's advanced Gateway system. In this example, we will be make a per member gateway with a built in key for points. We will be remaking the {@tutorial CreatingPointsSystems} example, in this tutorial.
+
+> Note: I would not recommend using the default (JSON) provider for this, as you may end up reaching the limit of JSON files your OS can open.
+
+## Creating The Gateway
+
+Inside your entry point (Where you define `new Client()`, for example, `index.js`) add this:
+
+```javascript
+client.gateways.register('members', {
+	experience: {
+		type: 'Integer',
+		default: 10
+	},
+	level: {
+		type: 'Integer',
+		default: 1
+	}
+}, { provider: 'rethinkdb' });
+```
+
+And that's it, we now have a built in members gateway, however we have no way of accessing them via the [GuildMember](https://discord.js.org/#/docs/main/master/class/GuildMember) class. We'll tackle that next.
+
+## Creating Our Own GuildMember Class
+
+We have the Gateway, so we just need to use a {@link Extendable} to use it properly. Make a new Extendable in /extendables and call it `GuildMember`. Add this in:
+
+```javascript
+const { Extendable } = require('klasa');
+
+module.exports = class extends Extendable {

```

Comment:
In my opinion, this should be a discord.js extension, not an extendable. RFC

Reply:
yeah, this should be a regular property like how the other Configuration instances are defined in klasa core. Meaning you have to use custom-structures from d.js